### PR TITLE
added unit tests for switchmap/dashboard/table/device.py

### PR DESCRIPTION
**Pull request by 404Nikhil**
Fixes: #242 

test logs:
image

Pull Request Summary: Add Unit Tests for Device Class

This pull request introduces comprehensive unit tests for the Device class located in switchmap/dashboard/table/device.py. These tests ensure the reliability and correctness of the interfaces and system methods by covering various edge cases and scenarios.
Key Features:

    Comprehensive Test Coverage:
        interfaces Method:
            Tests with valid data.
            Tests with empty data.
            Tests with None data.
        system Method:
            Tests with valid data.
            Tests with empty data.
            Tests with None data.

    Mocking Dependencies:
        Utilized unittest.mock.patch to mock external dependencies (interfaces.table and system.table).
        Ensured tests focus exclusively on the behavior of the Device class.

    Dynamic Path Configuration:
        Added logic to include the project root directory in sys.path, enabling proper module imports in the test environment.

    Reusable and Clean Test Structure:
        Used setUpClass and tearDownClass for class-level test setup and teardown.
        Incorporated consistent naming conventions and test assertions.

Benefits:

    Ensures the Device class behaves as expected under different input conditions.
    Improves codebase reliability and facilitates future development.
    Follows best practices for unit testing by isolating class logic from external dependencies.

## Summary by CodeRabbit

- **Tests**
	- Added a new test file for the Device class.
	- Implemented unit tests for interfaces and system methods.
	- Verified method behavior with various input conditions (valid, empty, and None data).

### CODE REVIEW

**Comment by palisadoes** [comment]
We have a policy of unassigning contributors who close PRs without getting validation from our reviewer team. This is because:

1. We start looking for people to review PRs when you submit them. 
2. We often contact them and link to the PR. If the PR is closed the whole effort is wasted.
3. The historical thread of reviewer comments is broken when the work is spread across multiple PRs. The quality of our code is affected negatively.

Please be considerate of our volunteers' limited time and our desire to improve our code base.

This policy is stated as a pinned post in all our Talawa repositories. Our YouTube videos explain why this practice is not acceptable to our Community.

In most cases you don’t have to close the PR to trigger the GitHub workflow to run again. Making a new commit and pushing it to your GitHub account will normally be sufficient.

Unfortunately, if this continues we will have to close the offending PR and unassign you from the issue.

File: tests/switchmap_/dashboard/table/test_device.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+
+import unittest
+import os
+import sys
+from unittest.mock import patch
+
+# Add the project root directory to sys.path
+ROOT_DIR = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "../../../../")
```
Comment (palisadoes) [inline tests/switchmap_/dashboard/table/test_device.py:10]: 1. This is going to fail in the workflows.
2. The script needs to be able to run from the root directory of the repository
3. Follow the other test  examples in the test directory.

**Review by palisadoes — CHANGES_REQUESTED** [review]

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Instead of using mocks, instantiate with the expected data so that you get expected results. Search the code base more carefully to see what real data is expected and its formatting.
2. This give false hope
